modules/kit-elements-defaults/assets/js/editor/hooks/data/fill-defaults-on-drop.js

```diff
@@ -15,19 +17,36 @@ export default class FillDefaultsOnDrop extends $e.modules.hookData.Dependency {
 		const { model } = args,
 			{ elType, widgetType } = model;
 
-		const allSettings = $e.data.cache.storage.getItem( 'kit-elements-defaults' ) || {},
-			elementSettings = allSettings[ widgetType || elType ];
+		const elementSettings = store.get( widgetType || elType );
 
-		if ( ! elementSettings ) {
+		if (
+			! elementSettings ||
+			'object' !== typeof elementSettings ||
+			! Object.keys( elementSettings ).length
+		) {
 			return true;
 		}
 
+		const settings = { ...( args.model.settings || {} ), ...elementSettings };
+
+		[ '__dynamic__', '__globals__' ].forEach( ( type ) => {
+			if (
+				! elementSettings[ type ] ||
+				typeof elementSettings[ type ] !== 'object' ||
+				! Object.keys( elementSettings.__globals__ ).length
```

Comment:
you're using this obscure condition twice. maybe extract to a function (`isPopulatedObject()` or something)?

Reply:
Done